### PR TITLE
add make targets for building release candidates

### PR DESCRIPTION
**Pull request by sdowell**
These make targets are intended to be used for tagging, building, and
publishing release candidates. Make targets for promoting release
candidates will be added in a subsequent change.

### CODE REVIEW

File: scripts/tag-release-candidate.sh [hunk 1]
```diff
@@ -79,9 +80,10 @@ if [ -z "$CS_VERSION" ]; then
   exit 1
 fi
 
-# If a user omits the leading "v" (enters 1.2.3 instead of v1.2.3), handle it.
-if ! [[ "${CS_VERSION}" = v* ]]; then
-  CS_VERSION="v${CS_VERSION}"
```
Comment (janetkuo) [inline scripts/tag-release-candidate.sh:84]: Why not keeping this?

File: scripts/tag-release-candidate.sh [hunk 2]
```diff
@@ -79,9 +80,10 @@ if [ -z "$CS_VERSION" ]; then
...
+# If a user omits the leading "v" (enters 1.2.3 instead of v1.2.3), throw an error.
+if ! [[ "${CS_VERSION}" =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+  echo "Error: CS_VERSION must be in the form v1.2.3 - got ${CS_VERSION}"
```
Comment (janetkuo) [inline scripts/tag-release-candidate.sh:85]: Combine this with L78-81 if we choose to throw an error here

File: scripts/tag-release-candidate.sh (hunk 1 above)
Comment (sdowell) [reply scripts/tag-release-candidate.sh:84]: IMO having a stricter check is preferred, as it leaves less room for error. The `__build` make target also builds on the assumption of the well formed version for finding the name of the RC tag.

The plan is for this to all be run via automation, so it shouldn't be a large inconvenience users.

File: scripts/tag-release-candidate.sh (hunk 2 above)
Comment (sdowell) [reply scripts/tag-release-candidate.sh]: I think having separate conditions and associated error messages is valid, it makes it clear whether the error was that
- The `CS_VERSION` was not provided
- The `CS_VERSION` was provided but is in an incorrect format

WDYT?

Comment (sdowell) [reply scripts/tag-release-candidate.sh]: I removed L78-81

**Review by janetkuo — APPROVED** [review]
/lgtm

**Pull request merged**